pkg/collector/metadata/agentchecks/agentchecks.go

```diff
@@ -11,6 +11,7 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/collector/autodiscovery"
 	"github.com/DataDog/datadog-agent/pkg/collector/runner"
 	"github.com/DataDog/datadog-agent/pkg/metadata/common"
+	"github.com/DataDog/datadog-agent/pkg/metadata/externalhost"
```

Comment:
I noticed the implementation for the externalhost cache doesn't look thread-safe.

Reply:
it's not indeed, good catch, will address in this PR since it shouldn't be much code